### PR TITLE
Make InputLayer support masking

### PR DESCRIPTION
**Pull request by yanboliang**
### Summary
```Sequential``` add ```InputLayer``` automatically if users don't assign. We ```Sequential``` was placed as downstream of ```Masking``` layer(#10763), it would throw exception due to ```InputLayer``` doesn't support masking.
I think ```InputLayer``` should support masking and it can use the default ```compute_mask``` function in base layer(just carry over the input mask). 

### Related Issues
#10763

### PR Overview

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: tests/keras/layers/core_test.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 from keras import backend as K
 from keras import layers
-from keras.models import Model
+from keras.models import Model, Sequential
```
Comment (fchollet) [inline tests/keras/layers/core_test.py]: One import per line

File: tests/keras/layers/core_test.py [hunk 2]
```diff
@@ -343,5 +343,30 @@ def test_activity_regularization():
     model.compile('rmsprop', 'mse')
 
 
+@keras_test
+def test_sequential_as_downstream_of_masking_layer():
+
+    input = layers.Input(shape=(3, 4))
```
Comment (fchollet) [inline tests/keras/layers/core_test.py]: `input` is a reserved Python keyword, use `inputs`
